### PR TITLE
Added Video Functionality

### PR DESCRIPTION
**Pull request by PrabhutvaAgrawal**
I have tried to add Video Functionality to the app by creating a new fragment. It will be able to play YouTube and FaceBook videos when video url is passed. Currently connenction to firebase is left and code is still incomplete.

### CODE REVIEW

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class VideosActivity extends AppCompatActivity {
+    private String videoID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_videos);
+        //getActionBar().hide();
+        videoID= getIntent().getStringExtra("id");
+        String dateStamp = getIntent().getStringExtra("date");
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java]: create a `public static` method in **`VideosActivity`** class to launch the activity with required parameters. (date, title, etc)

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.pierfrancescosoffritti.youtubeplayer.player.*;
+
+public class VideosActivity extends AppCompatActivity {
+    private String videoID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_videos);
+        //getActionBar().hide();
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/VideosActivity.java]: delete code that is not needed. Don't comment it out.

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/VideosAdapter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    public VideosViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        // view_g => name of the layout file
+        View view = inflater.inflate(R.layout.item_format_videos, parent, false);
+        VideosViewHolder holder = new VideosViewHolder(view);
+        return holder;
+    }//link xml to recycler view
+
+    @Override//means whatever we are extending is changed to put our own stuff
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/VideosAdapter.java]: lol, wkt

File: androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import static com.csatimes.dojma.activities.AboutDojmaActivity.TAG;
+import static com.csatimes.dojma.utilities.DHC.TAG_PREFIX;
+import static com.csatimes.dojma.utilities.DHC.getGridSpan;
+
+public class VideosFragment extends Fragment {
+
+    //private static final String TAG = TAG_PREFIX + VideosFragment.class.getSimpleName();
+    private RecyclerView videosRv;
+
+    String videoURL[] = {"https://www.facebook.com/DoPy.BITSGoa/videos/622644644796899/","https://www.facebook.com/bitsquark/videos/309257229884592/","https://www.facebook.com/bitsquark/videos/2300637253556600/","https://www.youtube.com/watch?v=IM2czqavlWM", "https://www.youtube.com/watch?v=737dCFBqO4I", "https://www.youtube.com/watch?v=yuboWIBPDvk", "https://www.youtube.com/watch?v=pXCpc2Bcirs", "https://www.youtube.com/watch?v=4AQm22YQfYU", "https://www.youtube.com/watch?v=3PtbK_0j1sw", "https://www.youtube.com/watch?v=WDIeS7NHTpU", "https://www.youtube.com/watch?v=LvetJ9U_tVY", "https://www.youtube.com/watch?v=6JYIGclVQdw"};
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java]: create a new class (YoutubeVideo, maybe) to hold these data members (url, title, etc)

File: androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    final Context context = getContext();
+    videosRv.setHasFixedSize(true);
+    videosRv.setLayoutManager(new LinearLayoutManager(context));
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        final Activity activity = getActivity();
+            final Context context = getContext();
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/fragments/VideosFragment.java]: use consistent alignment

**Comment by aryanagarwal15** [comment]
@tanmaydixit @Rushi98 Please merge this PR. I've completed everything required for the feature

**Comment by aryanagarwal15** [comment]
@tanmaydixit please merge!

**Pull request merged**